scene/2d/cpu_particles_2d.cpp

```diff
@@ -166,10 +166,20 @@ void CPUParticles2D::_update_mesh_texture() {
 	vertices.push_back(-tex_size * 0.5 + Vector2(tex_size.x, tex_size.y));
 	vertices.push_back(-tex_size * 0.5 + Vector2(0, tex_size.y));
 	PoolVector<Vector2> uvs;
-	uvs.push_back(Vector2(0, 0));
-	uvs.push_back(Vector2(1, 0));
-	uvs.push_back(Vector2(1, 1));
-	uvs.push_back(Vector2(0, 1));
+	AtlasTexture *at = cast_to<AtlasTexture>(*texture);
+	if (!at) {
+		uvs.push_back(Vector2(0, 0));
+		uvs.push_back(Vector2(1, 0));
+		uvs.push_back(Vector2(1, 1));
+		uvs.push_back(Vector2(0, 1));
+	} else {
+		Rect2 region_rect = at->get_region();
+		Size2 atlas_size = at->get_atlas()->get_size();
+		uvs.push_back(Vector2(region_rect.position.x/atlas_size.x, region_rect.position.y/atlas_size.y));
```

Comment:
This will fail code style checks, there should be spaces around math operators. See docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html

To update the PR, please do a `git commit --amend` and force push. See [PR workflow](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#modifying-a-pull-request).